astro/src/content/docs/extend/code/password-hashes/writing-a-plugin.mdx

```diff
@@ -9,15 +9,23 @@ import InlineField from 'src/components/InlineField.astro';
 import InlineUIElement from 'src/components/InlineUIElement.astro';
 import Aside from 'src/components/Aside.astro';
 import TroubleshootingPlugin from 'src/content/docs/extend/code/password-hashes/_troubleshooting-plugin.mdx';
-import {RemoteCode} from '@fusionauth/astro-components';
+import { RemoteCode } from '@fusionauth/astro-components';
 
 
-## Install Java 8
+## Install Java
 
-If you haven't already, you need to download the Java 8 JDK. This contains all of the tools you need to compile Java projects and create JAR files. You can download the Java 8 JDK from the Oracle at the following link.
+If you haven't already, you need to download the Java JDK. This contains all of the tools you need to compile Java projects and create JAR files. You can download the Java JDK from  Oracle at the following link.
 
 - https://www.oracle.com/java/technologies/javase-downloads.html
 
+<Aside type="note">
+For plugins the Java version must be 21 or lower. That is a hard requirement since the byte code for 23 will not run in 21.
+
+To provide maximum compatibility we recommend using Java at version 8.
+
+In a spec file this would be represented as `min java version is 8, max is 21`.
```

Comment:
I don't understand what a spec file is and don't see it mentioned elsewhere on this page. What is it?